esmvaltool/diag_scripts/carbon_cycle/main.ncl

```diff
@@ -0,0 +1,626 @@
+;                            automatically.
+;     evolution_plot_volcanoes: Turns on/off lines of volcano eruptions.
+;     evolution_plot_color: Hue of the contours; default: red = 0.
+;
+;     For external dataset input:
+;     obsfile: Filename of external obs to read in.
+;     obsname: Name of ext dataset to read in.
+;     ref_varname: Varname in attributes of ext file.
+;
+; Caveats:
+;     If CRU is the reference dataset it is important to apply a landmask in
+;     the preprocessor section of the recipe.
+;     Regridding is not required since spatial averages are calculated for all
+;     plots. Performance metrics calculated with perfmetrics_main could be
+;     inconsistent with the plots produced by this script, since a consistent
+;     missing value mask is not applied here. However, for variable such NBP,
+;     GPP, LAI, FGCO2 for which this script is designed, there should be no
+;     inconsistencies as the missing value mask is similar for all datasets
+;     and observations and coincides with the land-sea mask.
+;
+; Modification history:
+;     20180816_A-schl_ma: continued porting to v2.0.
+;     20180619_A-wenz_sa: adopted script to v2.
+;     20170330_A-gier_be: added external data-read in (e.g. GCP).
+;     20170316_A-gier_be: added cdo regridding, landmask for CRU obs.
+;     20170308_A-gier_be: added "evolution_plot".
+;     20151130_A-laue_ax: corrected passing of file type from namelist to
+;                         plotting functions; moved call to write_references
+;                         to beginning of code.
+;     20151105_A-righ_ma: graphic part moved to external plot_scripts.
+;     20151030_A-righ_ma: regridding not working on ocean grids, removed since
+;                         it is not necessary for these diagnostics.
+;     20151029_A-righ_ma: replacing area functions with ESMValTool functions.
+;     20151021_A-righ_ma: pep8 and header standardization.
+;     201510??_A-wenz_sa: adapted to ESMValTool structure.
+;     201505??_A-anav_al: written.
+;
+; #############################################################################
+
+load "$diag_scripts/../interface_scripts/interface.ncl"
+
+load "$diag_scripts/shared/latlon.ncl"
+load "$diag_scripts/shared/scaling.ncl"
+load "$diag_scripts/shared/set_operators.ncl"
+load "$diag_scripts/shared/statistics.ncl"
+
+load "$diag_scripts/shared/plot/scatterplot.ncl"
+load "$diag_scripts/shared/plot/style.ncl"
+load "$diag_scripts/shared/plot/xy_line.ncl"
+
+begin
+
+  enter_msg(DIAG_SCRIPT, "")
+  AUTHORS = (/"anav_alessandro", "wenzel_sabrina", "righi_mattia", \
+             "schlund_manuel"/)
+  REFERENCES = (/"anav13jclim"/)
+  DOMAIN = diag_script_info@region
+
+  ; Variable
+  VAR0 = variable_info[0]
+  var0 = VAR0@short_name
+  DIM_VAR = ListCount(variable_info)
+  err = False
+  if (DIM_VAR .eq. 2) then
+    if (variable_info[1]@short_name .ne. var0) then
+      err = True
+    end if
+  else if (DIM_VAR .gt. 2) then
```

Comment:
`else if` has been replaced by `elseif` in the newer NCL versions, which also removes the need for repeating `end if` at the end of the block.